lighthouse-cli/test/smokehouse/smokehouse-report.js

```diff
@@ -102,45 +102,74 @@ function findDifference(path, actual, expected) {
   return null;
 }
 
+/**
+ * @param {string} name
+ * @param {any} actualResult
+ * @param {any} expectedResult
+ * @returns {Smokehouse.Assertion}
+ */
+function makeAssertion(name, actualResult, expectedResult) {
+  const diff = findDifference(name, actualResult, expectedResult);
+
+  return {
+    category: name,
+    actual: actualResult,
+    expected: expectedResult,
+    equal: !diff,
+    diff,
+  };
+}
+
 /**
  * Collate results into comparisons of actual and expected scores on each audit.
- * @param {Smokehouse.ExpectedLHR} actual
- * @param {Smokehouse.ExpectedLHR} expected
- * @return {Smokehouse.LHRComparison}
+ * @param {Smokehouse.ExpectedRunResult} actual
+ * @param {Smokehouse.ExpectedRunResult} expected
+ * @return {Smokehouse.RunComparison}
  */
 function collateResults(actual, expected) {
-  const auditNames = Object.keys(expected.audits);
-  const collatedAudits = auditNames.map(auditName => {
-    const actualResult = actual.audits[auditName];
-    if (!actualResult) {
-      throw new Error(`Config did not trigger run of expected audit ${auditName}`);
-    }
+  /** @type {Smokehouse.Assertion[]} */
+  const artifactAssertions = [];
+  if (expected.artifacts) {
+    Object.keys(expected.artifacts).map(artifactName => {
+      const actualResult = /** @type {any} */ (actual.artifacts)[artifactName];
```

Comment:
can we type these as partial artifacts? I guess it doesn't help much since we don't type check our assertions :)

Reply:
Do artifacts have a type other than the union type of all artifact values? Is that what you were thinking of?

Reply:
Oh I get it now, we can type the artifacts key instead of it being string and then the artifact value will be the union type.

Reply:
Awesome yeah LGTM :)